README.md

```diff
@@ -107,7 +107,20 @@ Rest Server uses exactly the same directory structure as local backend, so you s
 
 ### Systemd
 
-There's an example [systemd service file](https://github.com/restic/rest-server/blob/master/examples/systemd/rest-server.service) included with the source, so you can get Rest Server up & running as a proper Systemd service in no time.  Before installing, adapt paths and options to your environment.
+There's an example [systemd service file][systemd-service-file] included with the source, so you can get Rest Server up & running as a proper Systemd service in no time.
+
+The service file can directly be copied (typically in `/usr/lib/systemd/system`) and customised before starting the service with for example:
+
+```sh
+sudo install -D -m644 examples/systemd/rest-server.service /usr/lib/systemd/system
+sudo EDITOR=$EDITOR systemctl edit rest-server --full
```

Comment:
> Wouldn't it be better to adjust the configuration before copying it to the system configuration.

Not if the service-file is deployed by a package. Packagers can not provide a `.service` file that works for everyone (they will have however to patch the example, typically for the binary path), and users will *not* re-patch that file to adjust the config: this is not the `.service` file purpose. The `systemctl edit` is a workaround because `rest-server` doesn't provide a way of customizing the configuration (for example through a config file, or for what it's worth, environment variables).

> Is a call to systemctl daemon-reload also necessary?

Quoting `systemctl(1)`:

> After the units have been edited, systemd configuration is reloaded (in a way that is equivalent to daemon-reload).

Reply:
I've just noticed that the issue description states "useful for casual users and packagers". However, I think that's a bad idea: for packaging a service e.g. different directories should be used compared to manually installing.  Depending on the system to package for there might even be different conventions how things should be done. Thus, in my opinion the readme should focus only on users.

> Not if the service-file is deployed by a package. Packagers can not provide a `.service` file that works for everyone (they will have however to patch the example, typically for the binary path), and users will _not_ re-patch that file to adjust the config: this is not the `.service` file purpose.

Adapting the configuration in the service file is nearly impossible to avoid when sandboxing is used. Or is there a way to let the `ReadWritePaths` depend on some external configuration file?

It is possible for a package to provide a configuration that at least works out of the box. A packaged version of rest-server won't call `systemctl edit` to make these necessary changes to the example configuration. 

>  The `systemctl edit` is a workaround because `rest-server` doesn't provide a way of customizing the configuration (for example through a config file, or for what it's worth, environment variables).

But then why use `systemctl edit --full`? It copies the whole service configuration instead of creating a drop-in that can specifically make system-specific adjustments.

Reply:
> I've just noticed that the issue description states "useful for casual users and packagers". However, I think that's a bad idea: for packaging a service e.g. different directories should be used compared to manually installing. Depending on the system to package for there might even be different conventions how things should be done. Thus, in my opinion the readme should focus only on users.

Alright.

Can we take a step back from the README change in that PR though: how would a packager build a package for rest-server?

Let's say they build a package distributing a working `rest-server.service` (because they create the corresponding system user, adjust the example to a working default path, etc). In that scenario, how is a user on that distribution supposed to customize it the path?

I'm asking because they can not customize the `--path` of the `ExecStart` using `systemctl edit` (without `--full` it adds a 2nd directing instead of replacing it).

Reply:
A package maintainer would probably just select `/var/lib/rest-server` as storage location.

The trick is to first clear the old command.
```
ExecStart=
ExecStart=/new/command
```

Or much cleaner use environment variables to configure the service options:
```
Environment=REPOPATH=/var/lib/rest-server
EnvironmentFile=-/etc/default/rest-server
ExecStart=rest-server --path $REPOPATH [...]
```
I'm not sure whether `ReadWritePaths=` also interprets environment variables which would be necessary to move the path configuration outside the service unit.

Reply:
I tried the approach you proposed but it doesn't work because ReadWritePaths doesn't accept variable from the environment. So I guess a packager would have to remove that entry?

Reply:
Depends on the packaging. In Debian quite a few services have the following section in their `EnvironmentFile`:
```
# WARNING: This file is used for compatibility with sysv init only.
# If you are using systemd (Debian default), please override some-service-name.service
# unit instead of modifying these variables.
```

Just removing `ReadWritePaths` won't work as that will render everything read-only due to `ProtectSystem=strict`. So the best way for a packager is probably to add a comment containing an example how to customize the repopath. It's of course also possible to remove more of the sandboxing, but that's somewhat comparable to just running everything as root (although removing the sandbox is by far not as bad as running everything as root)

Reply:
> So the best way for a packager is probably to add a comment containing an example how to customize the repopath.

This will imply patching the unit file or using `systemctl edit --full`, correct? Or do you see a way a config file could affect this?

Reply:
No. Just use `systemctl edit` to create a drop-in file to customize the service and add something like this:
```
[Service]
ExecStart=
ExecStart=/usr/local/bin/rest-server --path /path/to/backups
# maybe reset the sandbox paths before using `ReadWritePaths=`
ReadWritePaths=/path/to/backups
```

Reply:
OK I see.

I guess I'll close this PR for now then. Thanks.